### PR TITLE
Updating the readme with better instructions and swift development.

### PR DESCRIPTION
**Pull request by Atharva-Kanherkar**
## Proposed Changes
 

This PR addresses the need for more detailed instructions on setting up Node.js, handling errors, and configuring the Zowe environment on Windows. It includes information on:

- Choosing the right Node version.
- Installing Node on Windows without `nvm`.
- Common errors and their resolutions.
- Setting environment variables.
- Validating YAML configuration.
- Handling unhandled rejection errors and server issues.

This PR depends upon the following PRs: None

## Type of Change
 
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Change in documentation
- [ ] Refactor the code
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist

- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [x] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] Video or image is included if visual changes are made
- [ ] Relevant update to CHANGELOG.md
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

## Testing
 

The changes are documentation updates, so testing involves reviewing the added sections to ensure they are clear and provide accurate guidance for setting up Node.js, handling errors, and configuring the Zowe environment on Windows! 

## Further Comments
 
These updates are intended to provide more comprehensive support for users setting up and troubleshooting their Zowe environment on Windows, addressing common issues and enhancing the overall Development and setup process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,10 +157,35 @@ If you encounter an error message saying `No config file found, initializing`, i
...
+- If the install step still fails, try removing the `package-lock.json` entirely.
+
+
+### Customizing Configurations and Server Errors
+
+#### Validating YAML Configuration
+Ensure your `zowe.yaml` file is valid YAML. Use any online YAML checker to validate your file. Proper configuration is crucial for the system to function correctly. Please refer the information covered in [Section 7](#7-adding-zss-to-the-environment) to configure your file correctly.
+
+#### Handling Unhandled Rejection Errors
+**Known Issue & Workaround:** If you get an error stating `Unhandled rejection, Type error: Cannot read properties of undefined` after attempting to start your App Server, the issue is most likely with your configuration. Due to some highly customized configuration settings, if there is no easily identifiable cause, you can revert to a less strict version of the App Server via these two commands:
```
Comment (1000TurquoisePogs) [inline README.md]: This is a known issue? Where is the ticket describing it?
This reads more like

Problem: there is some bug
Workaround: reset everything

How about we learn more about this bug and try to fix it

File: README.md [hunk 2]
```diff
@@ -81,7 +81,17 @@ On z/OS, git 2.14.4 is the minimum needed.
...
 To use the App Server, the following is required:
 
-* **NodeJS** - v16.x up to v18.x is officially supported by the Zowe community.
+## Node Version Setup for Zowe on Windows
+
+### Choosing the Right Node Version
+For setting up Zowe, it's recommended to install Node 16.x. Although Zowe supports up to Node 18, Node 16.x is more stable and tends to work better without errors.
+
+### Installing Node on Windows
+Avoid using `nvm` for updating Node versions on Windows as it can cause bugs. Instead, delete your current version and install the exact version from the [Node.js official website](https://nodejs.org/).
```
Comment (1000TurquoisePogs) [inline README.md]: It can? like what?

File: README.md [hunk 3]
```diff
@@ -81,7 +81,17 @@ On z/OS, git 2.14.4 is the minimum needed.
...
+
+### Common `nvm` Errors on Windows
+Using `nvm` on Windows might result in errors like "Can't find the module ncp". It's better to avoid using `nvm` to prevent these issues.
```
Comment (1000TurquoisePogs) [inline README.md]: Sounds like a bug we should fix rather than telling people not to use nvm

**Review by 1000TurquoisePogs — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (DivergentEuropeans) [reply README.md]: It's the "Common nvm errors on Windows" area below @1000TurquoisePogs 

@Atharva-Kanherkar Let's bridge these two sections together, they don't need their own. Also I would prefer some of these to be "Notes"

File: README.md (hunk 1 above)
Comment (DivergentEuropeans) [reply README.md]: @Atharva-Kanherkar Make a ticket for this using the Zlux repo's "Issues" page. Lmk if you need help filling out any fields

@1000TurquoisePogs the fix for this we are unsure, how long will take. I played around with it for half a day, without success so it won't be trivial. We have multiple start-up/configuratio nrelated bugs. But I'll have them look into it

Comment (Atharva-Kanherkar) [reply README.md]: On it!

File: README.md (hunk 2 above)
Comment (Atharva-Kanherkar) [reply README.md]: Okay sure, would you like these to be notes under the node version or under the whole readme?

File: README.md (hunk 3 above)
Comment (Atharva-Kanherkar) [reply README.md]: Yes, I agree, but there were some problems troubleshooting, so can we keep it until we solve the bug? Or should i remove it?

**Review by DivergentEuropeans — APPROVED** [review]

**Pull request merged**